services/sts/src/main/java/software/amazon/awssdk/services/sts/auth/SessionCredentialsHolder.java

```diff
@@ -31,18 +31,51 @@ final class SessionCredentialsHolder {
     private final AwsSessionCredentials sessionCredentials;
     private final Date sessionCredentialsExpiration;
 
-    SessionCredentialsHolder(Credentials credentials) {
-        this.sessionCredentials = AwsSessionCredentials.create(credentials.accessKeyId(),
-                                                            credentials.secretAccessKey(),
-                                                            credentials.sessionToken());
+    SessionCredentialsHolder(Builder b) {
+        Credentials credentials = b.credentials;
+        this.sessionCredentials = AwsSessionCredentials.builder()
+                                                       .accessKeyId(credentials.accessKeyId())
+                                                       .secretAccessKey(credentials.secretAccessKey())
+                                                       .sessionToken(credentials.sessionToken())
+                                                       .accountId(b.accountId)
+                                                       .build();
         this.sessionCredentialsExpiration = Date.from(credentials.expiration());
     }
 
+    public static Builder builder() {
+        return new Builder();
+    }
+
     public AwsSessionCredentials getSessionCredentials() {
         return sessionCredentials;
     }
 
     public Date getSessionCredentialsExpiration() {
         return sessionCredentialsExpiration;
     }
```

Comment:
We should not be using `Date` for anything. `Instant`! (I know you didn't do this, but I'm shocked to see it here).

Reply:
Lol, yes I will update this. I kind of noticed it as I was refactoring the class but forgot it

Reply:
Oh wow it's an instant that we converted to a date and then back to instant again.